src/userobjects/MoabSkinner.C

```diff
@@ -0,0 +1,1008 @@
+        _id_to_elem_handles[id] = std::vector<moab::EntityHandle>();
+
+      _id_to_elem_handles[id].push_back(ent);
+
+      // Save the handle for adding to entity sets
+      all_elems.insert(ent);
+    }
+  }
+
+  // Add the elems to the full meshset
+  _moab->add_entities(_meshset, all_elems);
+
+  // Save the first elem
+  offset = all_elems.front();
+}
+
+const std::vector<std::vector<unsigned int>> &
+MoabSkinner::getTetSets(ElemType type) const
+{
+  if (type == TET4)
+    return _tet4_nodes;
+  else if (type == TET10)
+    return _tet10_nodes;
+  else
+    mooseError("The MoabSkinner can only be used with a tetrahedral [Mesh]!");
+}
+
+void
+MoabSkinner::createTags()
+{
+  // First some built-in MOAB tag types
+  _moab->tag_get_handle(GEOM_DIMENSION_TAG_NAME,
+                        1,
+                        moab::MB_TYPE_INTEGER,
+                        geometry_dimension_tag,
+                        moab::MB_TAG_DENSE | moab::MB_TAG_CREAT);
+
+  _moab->tag_get_handle(GLOBAL_ID_TAG_NAME,
+                        1,
+                        moab::MB_TYPE_INTEGER,
+                        id_tag,
+                        moab::MB_TAG_DENSE | moab::MB_TAG_CREAT);
+
+  _moab->tag_get_handle(CATEGORY_TAG_NAME,
+                        CATEGORY_TAG_SIZE,
+                        moab::MB_TYPE_OPAQUE,
+                        category_tag,
+                        moab::MB_TAG_SPARSE | moab::MB_TAG_CREAT);
+
+  _moab->tag_get_handle(NAME_TAG_NAME,
+                        NAME_TAG_SIZE,
+                        moab::MB_TYPE_OPAQUE,
+                        name_tag,
+                        moab::MB_TAG_SPARSE | moab::MB_TAG_CREAT);
+
+  // Some tags needed for DagMC
+  _moab->tag_get_handle("FACETING_TOL",
+                        1,
+                        moab::MB_TYPE_DOUBLE,
+                        faceting_tol_tag,
+                        moab::MB_TAG_SPARSE | moab::MB_TAG_CREAT);
+
+  _moab->tag_get_handle("GEOMETRY_RESABS",
+                        1,
+                        moab::MB_TYPE_DOUBLE,
+                        geometry_resabs_tag,
+                        moab::MB_TAG_SPARSE | moab::MB_TAG_CREAT);
+
+  // Set the values for DagMC faceting / geometry tolerance tags on the mesh entity set
+  _moab->tag_set_data(faceting_tol_tag, &_meshset, 1, &_faceting_tol);
+  _moab->tag_set_data(geometry_resabs_tag, &_meshset, 1, &_geom_tol);
+}
+
+void
+MoabSkinner::createGroup(unsigned int id, std::string name, moab::EntityHandle & group_set)
+{
+  // Create a new mesh set
+  _moab->create_meshset(moab::MESHSET_SET, group_set);
+
+  // Set the tags for this material
+  setTags(group_set, name, "Group", id, 4);
+}
+
+void
+MoabSkinner::createVol(unsigned int id,
+                          moab::EntityHandle & volume_set,
+                          moab::EntityHandle group_set)
+{
+  _moab->create_meshset(moab::MESHSET_SET, volume_set);
+
+  setTags(volume_set, "", "Volume", id, 3);
```

Comment:
I'm not sure exactly what you're suggesting - are you saying that "3" should be swapped with `GEOM_DIMENSION`?

Reply:
There's an additional tag that's typically set on `volume_set` that identifies the dimensionality what the elements in the mesh set are intended to represent. 

https://bitbucket.org/fathomteam/moab/src/645c5735a8fd264d68b9f8532b2b039fd350e127/src/MBTagConventions.hpp#lines-70

Reply:
It might be easier to elaborate on this offline. I can also have a look and make what I think is a mini-PR for this afterward to add this in if you prefer.

Reply:
Sure, I'm good with tackling this in a later PR. I'll add it to this catch-all issue (https://github.com/neams-th-coe/cardinal/issues/567) to make sure we keep track of it.